### PR TITLE
[WIP] Remove Relational.Design package

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #7004 
(first PR in series of PRs)
Refactoring/changing the architecture first to allow easy split of Relational.Design

### CODE REVIEW

**Comment by ErikEJ** [comment]
Providers beware? Does this mean move provider design bits into provider?

**Comment by smitpatel** [comment]
@ErikEJ - Yes. Provider.Design will go into Provider directly. Also reducing the code provider has to write.
This is first part of bigger change. Hope changes in SQLite.Design & SqlServer.Design would provider enough guidance to providers what to change.

File: src/EFCore.Design/Scaffolding/Internal/ReverseEngineeringGenerator.cs [hunk 1]
```diff
@@ -52,59 +53,69 @@ public class ReverseEngineeringGenerator
...
-                dbContextClassName, metadataModel, configuration.OverwriteFiles);
+            var @namespace = rootNamespace;
+            if (!string.Equals(fullOutputPath, fullProjectPath)
+                && fullOutputPath.StartsWith(fullProjectPath, StringComparison.Ordinal))
+            {
+                var relativeOutputPath = fullOutputPath.Substring(fullProjectPath.Length + 1);
+                @namespace += "." + string.Join(
+                                  ".", relativeOutputPath
+                                      .Split(_directorySeparatorChars, StringSplitOptions.RemoveEmptyEntries)
+                                      .Select(p => CSharpUtilities.Instance.GenerateCSharpIdentifier(p, null)));
```
Comment (bricelam) [inline src/EFCore.Design/Scaffolding/Internal/ReverseEngineeringGenerator.cs]: There is code that does this in Migrations too. (Added a comment to #1671)

File: src/EFCore.Design/Scaffolding/Internal/ReverseEngineeringGenerator.cs [hunk 2]
```diff
@@ -52,59 +53,69 @@ public class ReverseEngineeringGenerator
...
+            }
 
-            return CodeWriter.WriteCodeAsync(
-                modelConfiguration, outputPathsAndNamespace.CanonicalizedFullOutputPath,
-                dbContextClassName, cancellationToken);
-        }
+            //var customConfiguration = _configurationFactory.CreateCustomConfiguration(connectionString, contextName, @namespace, useDataAnnotations);
```
Comment (bricelam) [inline src/EFCore.Design/Scaffolding/Internal/ReverseEngineeringGenerator.cs]: Intentional?

Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/ReverseEngineeringGenerator.cs]: This is WIP. I m creating PRs against this branch and merging them to allow easy rebasing. #8583 will remove this line.

**Pull request merged**